src/helpers/permissions.js

```diff
@@ -0,0 +1,115 @@
+// Set of helper functions to help with the permissioning middlewares
+
+const url = require('url')
+const boom = require('boom')
+const roles = require('../constants/roles.js')
+const { rejectIfNull } = require('../db/utils')
+const { messages: errMessages } = require('../constants/errors.json')
+
+const Event = require('../models/Event')
+const Place = require('../models/Place')
+const Product = require('../models/Product')
+const User = require('../models/User')
+
+const orderedRoles = [roles.BASIC, roles.ADMIN, roles.SUPER]
+
+const hasSufficientRole = ({ minRole }) => user => {
+  /**
+  * Check's whether user's role is equal or higher than the minRole
+  *
+  * returns Boolean
+  *
+  * throws error on initialisation if passed option is not valid
+  */
+
+  if (!orderedRoles.includes(minRole)) {
+    throw boom.badImplementation()
+  }
+
+  return user && (orderedRoles.indexOf(user.role) >= orderedRoles.indexOf(minRole))
+}
+
+const checkUserOwnsResource = resourceType => resourceId => user => {
+  /**
+  * Checks Ownership of a resource
+  *
+  * returns a Promise that resolves to true or false, or rejects if
+  * resourceId is not found in DB (404)
+  *
+  * N.B. when the resource type itself is a User, the method is
+  * different but the returns are the same
+  */
+
+  if (resourceType === User) {
+    return Promise.resolve(resourceId === user.id.toString())
+  }
+
+  return resourceType.findById(resourceId)
+  .then(rejectIfNull(errMessages.GET_ID_NOT_FOUND))
+  .then(doc => {
```

Comment:
one line?

Reply:
yeah this was one line originally then I had a bug and forgot to change it back